src/main/kotlin/no/nav/klage/oppgave/clients/OppgaveClient.kt

```diff
@@ -59,7 +60,7 @@ class OppgaveClient(
             .header("Nav-Consumer-Id", applicationName)
             .retrieve()
             .bodyToMono<OppgaveResponse>()
-            .block() ?: throw RuntimeException("Oppgaver could not be fetched")
```

Comment:
Tror det er tre (minst) resultater vi må skille på.

- Bruker ber om én oppgave som ikke finnes (404)
- Bruker henter/søker en liste av oppgaver, men er ingen oppgaver for tiden (200)
- Kallet mot det bakomliggende oppgave-apiet (gosys) feiler (500)

F.eks. i dette eksemplet så tror jeg exception kastes hvis "gosys" feiler, og vi ikke får en respons tilbake.

Reply:
Jeg tenkte litt på det du skriver om her, og syntes ikke det var helt rett frem.. Jeg ville tro at hvis man ber om en oppgave som ikke finnes, så burde Oppgave-systemet returnere 404, ikke tom response. Tilsvarende, hvis man henter en liste med oppgaver, så burde man en tom liste, ikke en tom response. Og hvis Oppgave feiler, så burde det returnere en 500-feil, ikke en tom response. Så når er det egentlig bodyToMono returnerer en tom response?

Reply:
Samtidig så tenker jeg vel kanskje at det du tar opp her ikke er blitt hverken bedre eller dårligere ifm denne PRen, så kanskje det kan håndteres i en egen PR (uavhengig av hva man tenker gjøre)?

Reply:
`block()` vil nok "aldri" returnere `null` nei. Enten får vi "riktig" data eller så kastes en exception.
 
Så ingen av disse vil slå til tror jeg. På den måten så har nok inget blitt bedre eller dårligere enn før. Muligens blir det misvisende hvis man leser koden og tror det skal bety 404 når det skjer. Det stemmer jo ikke når denne exception blir kastet.